### PR TITLE
Update Nabu plugin to build two well-defined input formats, instead of relying on remotejson configuration

### PR DESCRIPTION
**Pull request by callunity**
JIRA Ticket: GP-4094

- [x] Updates Changelog
- [x] Updates developer documentation

These new classes will be reused when developing the Shesmu CreateArchive action.

### CODE REVIEW

**Comment by callunity** [comment]
Would love to hear thoughts on if `NabuCaseArchiveFormatDefinition` is too wordy, and if that should just be `CaseArchiveFormatDefinition` (with similar dropping of the `Nabu` prefix for other Dtos and Value classes in the package)

**Comment by avarsava** [comment]
> Would love to hear thoughts on if `NabuCaseArchiveFormatDefinition` is too wordy, and if that should just be `CaseArchiveFormatDefinition` (with similar dropping of the `Nabu` prefix for other Dtos and Value classes in the package)

I feel if you're already in the Nabu plugin directory then you know what you're working with and it doesn't need to be specified, but on the other hand the Pinery plugin prefixes every single file with 'Pinery' already

File: plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                    ca.getWorkflowRunIdsForVidarrArchival() == null
+                        ? Optional.empty()
+                        : Optional.of(ca.getWorkflowRunIdsForVidarrArchival()));
+              });
+    }
+
+    @Override
+    protected Stream<NabuCaseArchiveValue> fetch(Instant lastUpdated) throws Exception {
+      if (config.isEmpty()) {
+        return Stream.empty();
```
Comment (avarsava) [inline plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java:66]: please `return new ErrorableStream<>(Stream.empty(), false);` in these cases

File: plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPluginType.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  @Override
+  public NabuPlugin create(Path filePath, String instanceName, Definer<NabuPlugin> definer) {
+    return new NabuPlugin(filePath, instanceName, definer);
+  }
+
+  @Override
+  public void writeJavaScriptRenderer(PrintStream writer) {
+    // do nothing right now
```
Comment (avarsava) [inline plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPluginType.java:24]: This method has an empty implementation in PluginFileType so this method can be removed entirely if you don't intend to implement this

File: plugin-nabu/README.md [hunk 3]
```diff
@@ -1,22 +1,15 @@
 # Nabu
 [Nabu](https://github.com/oicr-gsi/nabu) is a web application which tracks the QC
 status of files, as well as the archive status/progress for cases.
-The Nabu plugin for Shesmu provides Nabu data for file QCs (`nabu` input source)
-and case archive status (`case_archive` input source).
 
-The input sources can be set up as follows:
+The Nabu plugin provides two input formats:
+* `nabu_file_qc` contains FileQC information
+* `case_archive` contains information about a case archive's archiving status
```
[the remark this answers is not in the document]
Comment (callunity) [reply plugin-nabu/README.md:7]: Pretty much, yeah - "sent to archiving dir", "backed up offsite", "sent to vidarr-archival", "files deleted"

[the remark this answers is not in the document]
Comment (callunity) [reply plugin-nabu/README.md:7]: (with timestamps/lack thereof to indicate where in the process a given case archive is)

File: plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPluginType.java (hunk 2 above)
Comment (callunity) [reply plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPluginType.java:24]: The plan is to eventually implement a CreateArchive action, which will implement this.

File: plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java (hunk 1 above)
Comment (callunity) [reply plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java]: done

File: plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        throws IOException, InterruptedException {
+      return HTTP_CLIENT
+          .send(
+              HttpRequest.newBuilder(URI.create(baseUrl + "/cases")).GET().build(),
+              new JsonListBodyHandler<>(MAPPER, NabuCaseArchiveDto.class))
+          .body()
+          .get()
+          .map(
+              ca -> {
+                return new NabuCaseArchiveValue(
```
[the remark this answers is not in the document]
Comment (callunity) [reply plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/NabuPlugin.java]: updated

**Review by avarsava — APPROVED** [review]

**Review by mlaszloffy — APPROVED** [review]
